v2/protocol/pubsub/protocol.go

```diff
@@ -160,17 +160,27 @@ func (t *Protocol) getOrCreateConnection(ctx context.Context, topic, subscriptio
 	return conn
 }
 
-// Receive implements Receiver.Receive
-func (t *Protocol) Receive(ctx context.Context) (binding.Message, error) {
+// Respond implements Responder.Respond. This response function doesn't support reply.
```

Comment:
Rather than do this, I would revert back to what you had, but change the channel:

```incoming chan pubsub.Message```

to be a Message and an Error, and there you can see if it should be a ACK or NACK from that error.

Reply:
Thanks for the pointer!